pkg/registry/apps/statefulset/storage/storage.go

```diff
@@ -17,31 +17,54 @@ limitations under the License.
 package storage
 
 import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/generic"
 	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
 	"k8s.io/apiserver/pkg/registry/rest"
 	"k8s.io/kubernetes/pkg/api"
-	appsapi "k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	extvalidation "k8s.io/kubernetes/pkg/apis/extensions/validation"
 	"k8s.io/kubernetes/pkg/registry/apps/statefulset"
 	"k8s.io/kubernetes/pkg/registry/cachesize"
 )
 
-// rest implements a RESTStorage for replication controllers against etcd
+// StatefulSetStorage includes dummy storage for StatefulSets and for Scale subresource.
```

Comment:
I implemented the framework based on replicaset. There is a similar comment for [replicaset implementation](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/extensions/replicaset/storage/storage.go#L38)

I guess the original author means `ReplicaSetStorage` encapsulates `REST`, `StatusREST` and `Registry`.